src/core/core_utils.js

```diff
@@ -90,11 +89,13 @@ function getInheritableProperty({
   getArray = false,
   stopWhenFound = true,
 }) {
-  const LOOP_LIMIT = 100;
-  let loopCount = 0;
   let values;
+  const visited = new Set();
 
-  while (dict) {
+  while (isDict(dict) && !visited.has(dict.objId)) {
```

Comment:
```suggestion
  while (dict instanceof Dict && !(dict.objId && visited.has(dict.objId))) {
```

Reply:
Why not `isDict` ? (I don't care but just to understand)

Reply:
Sorry, it probably doesn't matter here given that this code shouldn't be that hot.

---

However, in e.g. some parts of the `Parser`/`Lexer` and for some complex PDF documents, there's been cases where inlining these kind of checks and thus being guaranteed to skip unnecessary function calls have helped performance.